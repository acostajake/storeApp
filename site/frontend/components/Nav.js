import Link from 'next/link';

import NavStyles from './styles/NavStyles';

const Nav = () => (
    <NavStyles>
        <Link href='/store'>
            <a>Store</a>
        </Link>
        <Link href='/sell'>
            <a>Sell</a>
        </Link>
        <Link href='/signup'>
            <a>Signup</a>
        </Link>
        <Link href='/orders'>
            <a>Orders</a>
        </Link>
        <Link href='/profile'>
            <a>Profile</a>
        </Link>
    </NavStyles>
);

export default Nav;