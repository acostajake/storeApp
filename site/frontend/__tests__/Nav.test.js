import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import wait from 'waait';

import {
	signedInWithCartMocks,
	notSignedInMocks,
	signedInMocks,
} from '../lib/testUtils';
import Nav from '../components/Nav';

describe('<Nav/>', () => {
	it('renders a simple Nav when not logged in', async () => {
		const notLoggedInNav = mount(
			<MockedProvider mocks={notSignedInMocks}>
				<Nav />
			</MockedProvider>
		);
		await wait();
		notLoggedInNav.update();
		const nav = notLoggedInNav.find('ul[data-test="nav"]');
		expect(toJSON(nav)).toMatchSnapshot();
	});
	it('renders a simple Nav when not logged in', async () => {
		const loggedInNav = mount(
			<MockedProvider mocks={signedInMocks}>
				<Nav />
			</MockedProvider>
		);
		await wait();
		loggedInNav.update();
		const navBar = loggedInNav.find('ul[data-test="nav"]');
		expect(toJSON(navBar)).toMatchSnapshot();
	});
	it('renders the amount of items in a cart', async () => {
		const loggedInNav = mount(
			<MockedProvider mocks={signedInWithCartMocks}>
				<Nav />
			</MockedProvider>
		);
		await wait();
		loggedInNav.update();
		const count = loggedInNav.find('div.count');
		const nav = loggedInNav.find('[data-test="nav"]');
		expect(count.text()).toBe('9');
		expect(nav.children().length).toBe(7);
		expect(nav.last().text()).toContain('Sign Out');
	});
});
