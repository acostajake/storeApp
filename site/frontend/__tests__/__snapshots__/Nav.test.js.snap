// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> renders a simple Nav when not logged in 1`] = `
<ul
  className="NavStyles-sc-1l9s7xq-0 bWZogn"
  data-test="nav"
>
  <Link
    href="/store"
  >
    <a
      href="/store"
      onClick={[Function]}
    >
      Store
    </a>
  </Link>
  <Link
    href="/signup"
  >
    <a
      href="/signup"
      onClick={[Function]}
    >
      Log in / Sign Up
    </a>
  </Link>
</ul>
`;

exports[`<Nav/> renders a simple Nav when not logged in 2`] = `
<ul
  className="NavStyles-sc-1l9s7xq-0 bWZogn"
  data-test="nav"
>
  <Link
    href="/store"
  >
    <a
      href="/store"
      onClick={[Function]}
    >
      Store
    </a>
  </Link>
  <Link
    href="/sell"
  >
    <a
      href="/sell"
      onClick={[Function]}
    >
      Sell
    </a>
  </Link>
  <Link
    href="/signup"
  >
    <a
      href="/signup"
      onClick={[Function]}
    >
      Signup
    </a>
  </Link>
  <Link
    href="/orders"
  >
    <a
      href="/orders"
      onClick={[Function]}
    >
      Orders
    </a>
  </Link>
  <Signout>
    <Mutation
      mutation={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "SIGNOUT_MUTATION",
              },
              "operation": "mutation",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "signout",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "message",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 60,
            "start": 0,
          },
        }
      }
      refetchQueries={
        Array [
          Object {
            "query": Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": undefined,
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "me",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "email",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "permissions",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "orders",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "cart",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "quantity",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "item",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "id",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "price",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "image",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "title",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "desc",
                                          },
                                          "selectionSet": undefined,
                                        },
                                      ],
                                    },
                                  },
                                ],
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 189,
                "start": 0,
              },
            },
          },
        ]
      }
    >
      <button
        onClick={[Function]}
      >
        Sign Out
      </button>
    </Mutation>
  </Signout>
  <button
    onClick={[Function]}
  >
    My Cart
    <CartCount
      count={0}
    >
      <CartCount__AnimatedStyles>
        <span
          className="CartCount__AnimatedStyles-y5h8qq-0 ggBEpl"
        >
          <TransitionGroup
            childFactory={[Function]}
            component="div"
          >
            <div>
              <CSSTransition
                className="count"
                classNames="count"
                in={true}
                key=".$0"
                onExited={[Function]}
                timeout={
                  Object {
                    "enter": 600,
                    "exit": 600,
                  }
                }
                unmountOnExit={true}
              >
                <Transition
                  appear={false}
                  className="count"
                  enter={true}
                  exit={true}
                  in={true}
                  mountOnEnter={false}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  timeout={
                    Object {
                      "enter": 600,
                      "exit": 600,
                    }
                  }
                  unmountOnExit={true}
                >
                  <CartCount__Dot
                    className="count"
                  >
                    <div
                      className="CartCount__Dot-y5h8qq-1 gwbOBa count"
                    >
                      0
                    </div>
                  </CartCount__Dot>
                </Transition>
              </CSSTransition>
            </div>
          </TransitionGroup>
        </span>
      </CartCount__AnimatedStyles>
    </CartCount>
  </button>
</ul>
`;
